apps/builder/app/builder/features/style-panel/sections/backgrounds/background-parser.ts

```diff
@@ -0,0 +1,120 @@
+import * as csstree from "css-tree";
+import { colord } from "colord";
+import { parseCssValue } from "../../shared/parse-css-value";
+import type {
+  InvalidValue,
+  LayersValue,
+  RgbValue,
+  UnparsedValue,
+} from "@webstudio-is/css-data";
+
+export const gradientNames = [
+  "conic-gradient",
+  "linear-gradient",
+  "radial-gradient",
+  "repeating-conic-gradient",
+  "repeating-linear-gradient",
+  "repeating-radial-gradient",
+];
+
+export const parseBackground = (
+  background: string
+): {
+  backgroundImage: LayersValue | InvalidValue;
+  backgroundColor: RgbValue | undefined;
+} => {
+  const layers: string[] = [];
+
+  let tokenStream = background.trim();
+
+  tokenStream = tokenStream.endsWith(";")
```

Comment:
is there no way to use csstree to remove the properties? to avoid parsing mistakes mainly

Reply:
Haven't found, in most cases it produces ast from any input, but for some reason fails on ';'